go/vt/vtgate/planbuilder/select.go

```diff
@@ -136,22 +141,27 @@ func (pb *primitiveBuilder) processSelect(sel *sqlparser.Select, outer *symtab)
 	return nil
 }
 
-func tryAtVtgate(sel *sqlparser.Select) engine.Primitive {
+func handleDualSelects(sel *sqlparser.Select, vschema ContextVSchema) (engine.Primitive, error) {
 	if !isOnlyDual(sel.From) {
-		return nil
+		return nil, nil
 	}
 
 	exprs := make([]evalengine.Expr, len(sel.SelectExprs))
 	cols := make([]string, len(sel.SelectExprs))
 	for i, e := range sel.SelectExprs {
 		expr, ok := e.(*sqlparser.AliasedExpr)
 		if !ok {
-			return nil
+			return nil, nil
 		}
 		var err error
+		if sqlparser.IsLockingFunc(expr.Expr) {
+			// if we are using any locking functions, we bail out here and send the whole query to a single destination
+			return buildLockingPrimitive(sel, vschema)
```

Comment:
I'm wondering if we can enforce this construct to be simple. I just tried this:
```
mysql> select get_lock('aaa', 2);
+--------------------+
| get_lock('aaa', 2) |
+--------------------+
|                  1 |
+--------------------+
1 row in set (0.00 sec)

mysql> select get_lock('aaa', 2) from a.a;
Empty set (0.00 sec)
```

But if it's a pain, I'm ok with leaving this as is and tightening it later.

Reply:
That is next on the list for get_lock. We will get the get_lock PR merged and work on it next.